[CI] FlattenedUnmappedFieldsTests class failing

**Build Scans:**
- [elasticsearch-periodic #12603 / release-tests](https://gradle-enterprise.elastic.co/s/c2dhbgzpxdld4)
- [elasticsearch-periodic #12603 / release-tests](https://gradle-enterprise.elastic.co/s/vclf2snnlddrc)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.flattened.FlattenedUnmappedFieldsTests.testSettingRejectedOutsideColumnarMode" -Dtests.seed=CF17532094198975 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=sv-FI -Dtests.timezone=America/St_Kitts -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.flattened.FlattenedUnmappedFieldsTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.index.mapper.flattened.FlattenedUnmappedFieldsTests'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "only permitted in strict columnar index modes"
     but: was "unknown setting [index.mapping.flattened_unmapped_fields.enabled] please check that any required plugins are installed, or check the breaking changes documentation for removed settings"
```

**Issue Reasons:**
- [main] 1 consecutive failures in class org.elasticsearch.index.mapper.flattened.FlattenedUnmappedFieldsTests
- [main] 2 consecutive failures in step release-tests
- [main] 1 consecutive failures in pipeline elasticsearch-periodic
- [main] 2 failures in class org.elasticsearch.index.mapper.flattened.FlattenedUnmappedFieldsTests (8.3% fail rate in 24 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-core-infra.

:eyes:

--- a/server/src/test/java/org/elasticsearch/index/mapper/flattened/FlattenedUnmappedFieldsTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/flattened/FlattenedUnmappedFieldsTests.java
@@ -23,6 +23,7 @@ import org.elasticsearch.index.mapper.MapperServiceTestCase;
 import org.elasticsearch.index.mapper.ParsedDocument;
 import org.elasticsearch.simdvec.ESVectorUtil;
 import org.elasticsearch.xcontent.XContentBuilder;
+import org.junit.Before;
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
@@ -70,6 +71,11 @@ public class FlattenedUnmappedFieldsTests extends MapperServiceTestCase {
             }
         }
         return false;
+    }
+
+    @Before
+    public void testFeatureFlag() {
+        assumeTrue("flattened_unmapped_fields is enabled", FlattenedFieldMapper.UNMAPPED_FIELDS_FEATURE_FLAG.isEnabled());
     }
 
     public void testSettingRejectedOutsideColumnarMode() {

